src/ansys/simai/core/data/model_configuration.py

```diff
@@ -0,0 +1,257 @@
+                ModelConfiguration,
+                ModelInput,
+                ModelOutput,
+            )
+
+            simai = asc.from_config()
+
+            # Get the project of interest
+            aero_dyn_project = simai.projects.get(name="aero-dyn")
+
+            # Define the inputs of the model
+            model_input = ModelInput(surface=["Velocity"], boundary_conditions=["Vx"])
+
+            # Define the outputs of the model
+            model_output = ModelOutput(
+                surface=["Pressure", "WallShearStress_0"], volume=["Velocity_0", "Pressure"]
+            )
+
+            # Define the model coefficient
+            global_coefficients = [("max(Pressure)", "maxpress")]
+
+            # Define the build configuration for the model
+            new_conf = ModelConfiguration(
+                project=aero_dyn_project,
+                build_preset="debug",
+                continuous=False,
+                input=model_input,
+                output=model_output,
+                global_coefficients=global_coefficients,
+                simulation_volume=simulation_volume,
+            )
+
+            # Launch a mode build with the new configuration
+            new_model = simai.models.build(new_conf)
+    """
+
+    project: "Project" = None
+    build_preset: Literal[
+        "debug",
+        "short",
+        "default",
+        "long",
+    ] = "default"
+    continuous: bool = False
+    input: ModelInput = ModelInput()
+    output: ModelOutput = ModelOutput()
+
+    def __set_gc(self, gcs: list[tuple | GlobalCoefficientUnit]):
+        verified_gcs = []
+        for gc in gcs:
+            gc_unit = GlobalCoefficientUnit(*gc) if isinstance(gc, tuple) else gc
+            self.project.verify_gc_formula(
+                gc_unit.formula, self.input.boundary_conditions, self.output.surface
+            )
+            verified_gcs += [gc_unit]
+        self.__dict__["global_coefficients"] = verified_gcs
+
+    def __get_gc(self):
+        return self.__dict__.get("global_coefficients")
+
+    global_coefficients: ModelOutput = property(__get_gc, __set_gc)
+
+    def __init__(
+        self,
+        boundary_conditions: Optional[dict[str, Any]] = None,
+        build_preset: Optional[str] = None,
+        continuous: Optional[bool] = False,
+        fields: Optional[dict[str, Any]] = None,
+        global_coefficients: Optional[dict[dict[str, Any]] | GlobalCoefficientUnit] = None,
+        simulation_volume: Optional[dict[str, Any]] = None,
+        project: Optional["Project"] = None,
+        input: Optional[ModelInput] = None,
+        output: Optional[ModelOutput] = None,
+    ):
+        """Sets the properties of a build configuration."""
+        self.project = project
+        if boundary_conditions is not None:
+            self.input.boundary_conditions = list(boundary_conditions.keys())
+        self.build_preset = build_preset
+        self.continuous = continuous
+        if fields is not None:
```

Comment:
fields is not documented

Reply:
IMO, `fields` should not be documented, as they are not supposed to be used by clients/customers.